### ISSUE TITLE
Data Quality Step Fails After Following Workshop

### ISSUE THREAD

**Issue by piers-walter-ibm** [awslabs/aws-serverless-data-lake-framework#117]
**Describe the bug**
We have followed the [workshop](https://catalog.us-east-1.prod.workshops.aws/workshops/501cb14c-91b3-455c-a2a9-d0a21ce68114/en-US) and steps A & B run as expected, the data quality step fails however at the Glue job. The error message we receive is `Exception in User Class: java.lang.NoSuchMethodError : scala.Product.$init$(Lscala/Product;)V`. This happens on both the `sdlf-data-quality-profile-runner` Glue job and the `sdlf-data-quality-suggestion-analysis-verification-runner` job. 

**To Reproduce**
1. Deploy framework as in workshop
2. Deploy new data to the S3 bucket
3. Wait for data quality glue jobs to be triggered
4. See error in glue job runs console

**Expected behavior**
The data quality glue jobs should execute as intended

**SDLF release (if known):**
Unknown, but installed following workshop instructions last week (week starting 6th Feb 2023)

**Comment by piers-walter-ibm** [comment]
One step closer, we managed to get the `sdlf-data-quality-profile-runner` Glue job running by changing the Glue version to 3.0 and updating the deequ jar from `deequ-1.2.2-spark-2.4` to `deequ-2.0.0-spark-3.1`. 

`sdlf-data-quality-suggestion-analysis-verification-runner` is still failing but this time with a new error. This time in the [`writeToDynamoDB` function](https://github.com/awslabs/aws-serverless-data-lake-framework/blob/main/sdlf-foundations/scripts/deequ/resources/deequ-suggestion-analysis-verification-runner.scala#L147). This may have something to do with HDFS support being removed in Glue 3.0 but further investigation is needed

**Comment by piers-walter-ibm** [comment]
Further update, commenting out the line `writeToDynamoDB(suggestionDataFrameRenamed, dynamodbSuggestionTableName)` in `deequ-suggestion-analysis-verification-runner.scala` has allowed the pipeline to run as expected. It's not particularly clear to me why the data needs to be stored in DynamoDB but I'm sure there's a good reason. This stage ran successfully with the newer jar file as detailed in the comment above on Glue 3.0

**Comment by cnfait** [comment]
Thanks for reporting the issue. I've been able to reproduce it.

Change: https://github.com/awslabs/aws-serverless-data-lake-framework/commit/5065cd56f6b54cd3b4b65565c59c3ec0e4732f31

While using newer versions appears to work with minimal changes I've decided to go back to deequ 1.1.0 (instead of 1.2.2 with the issue) to avoid changing the Glue version (we will use later versions of deequ and Glue eventually, but only after more testing on our side).
